### PR TITLE
replace usage of v1beta1 VolumeAttachments with v1

### PR DESCRIPTION
**Pull request by andrewsykim**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
As part of CSI post-GA clean up

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/assign @msau42 @vladimirvivien

### CODE REVIEW

**Comment by msau42** [comment]
Does this [etcd test](https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/test/integration/etcd/data.go#L295) also need to be updated? 

Also this [e2e test](https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/test/e2e/storage/csi_volumes.go#L208)

**Comment by andrewsykim** [comment]
For `test/integration/etcd/data.go`, I wasn't sure if we should be removing the v1alpha1/v1beta1 test cases because the PR removes callers but doesn't actually remove the API versions.

**Comment by msau42** [comment]
> For `test/integration/etcd/data.go`, I wasn't sure if we should be removing the v1alpha1/v1beta1 test cases because the PR removes callers but doesn't actually remove the API versions.

I don't think we should remove the API versions, but is the test defaulting to v1beta?

**Comment by andrewsykim** [comment]
Ooh I see now, I'll update that, thanks!

File: pkg/kubeapiserver/default_storage_factory_builder.go [hunk 1]
```diff
@@ -54,7 +54,7 @@ func NewStorageFactoryConfig() *StorageFactoryConfig {
 		DefaultResourceEncoding: serverstorage.NewDefaultResourceEncodingConfig(legacyscheme.Scheme),
 		ResourceEncodingOverrides: []schema.GroupVersionResource{
 			batch.Resource("cronjobs").WithVersion("v1beta1"),
-			apisstorage.Resource("volumeattachments").WithVersion("v1beta1"),
+			apisstorage.Resource("volumeattachments").WithVersion("v1"),
```
Comment (andrewsykim) [inline pkg/kubeapiserver/default_storage_factory_builder.go]: Curious if this is still needed

Comment (liggitt) [reply pkg/kubeapiserver/default_storage_factory_builder.go]: v1 is the preferred version, you can drop this line completely

File: test/integration/etcd/data.go [hunk 2]
```diff
@@ -296,22 +296,22 @@ func GetEtcdStorageData() map[schema.GroupVersionResource]StorageData {
...
-			ExpectedGVK:      gvkP("storage.k8s.io", "v1beta1", "VolumeAttachment"),
+			ExpectedGVK:      gvkP("storage.k8s.io", "v1", "VolumeAttachment"),
 		},
 		// --
 
 		// k8s.io/kubernetes/pkg/apis/storage/v1beta1
 		gvr("storage.k8s.io", "v1beta1", "volumeattachments"): {
 			Stub:             `{"metadata": {"name": "va2"}, "spec": {"attacher": "gce", "nodeName": "localhost", "source": {"persistentVolumeName": "pv2"}}}`,
 			ExpectedEtcdPath: "/registry/volumeattachments/va2",
+			ExpectedGVK:      gvkP("storage.k8s.io", "v1", "VolumeAttachment"),
```
Comment (liggitt) [inline test/integration/etcd/data.go]: this is what verifies the data hitting etcd is in the version we expect, and this looks good

**Comment by andrewsykim** [comment]
PTAL again, had to resolve a conflict

**Comment by roycaihw** [comment]
cc @cheftako

**Pull request merged**